backend/ranking.js

```diff
@@ -5,38 +5,259 @@ import { Dislike } from "./models/dislike.js";
 import { compareStocksByPoints } from "./utils.js";
 import { User } from "./models/user.js";
 import { Friend } from "./models/friend.js";
+import { Ranking } from "./models/ranking.js";
 
 const FOLLOW_POINTS = 25;
 const LIKE_POINTS = 20;
 const DISLIKE_POINTS = -20;
 const FOLLOW_FRIEND_POINTS = 10;
 const LIKE_FRIEND_POINTS = 10;
 const DISLIKE_FRIEND_POINTS = -10;
+const MAX_PAGE_SIZE = 10;
 
-export async function RankingV3(user) {
+export async function RankingV5(user, page) {
     try {
-        const [stocksInDatabase, follows, likes, dislikes, connections] = await Promise.all([
-            Stock.findAll(),
-            Follow.findAll({
-                where: { UserId: user.id },
-                include: [{ model: Stock }]
-            }),
-            Like.findAll({
-                where: { UserId: user.id },
-                include: [{ model: Stock }]
-            }),
-            Dislike.findAll({
-                where: { UserId: user.id },
-                include: [{ model: Stock }]
-            }),
-            Friend.findAll({
-                where: { UserId1: user.id },
-                include: [
-                    { model: User, as: "user1" },
-                    { model: User, as: "user2" }
-                ]
+        if (page != 1) {
```

Comment:
The ranking algorithm should not be concerned (nor aware of) this logic, and vice versa.
These are two parts that can be decoupled and independent, yet work together.

One part/module knows about the page, about the database and decides if it wants to respond with what is stored, or compute a new ranking response, in which case it calls the ranking algorithm, store the result, and respond with it.

The other part is the algorithm, it should work the same without any change from the previous version. 
I think version 3 should work actually.